### PR TITLE
cmake: Use variables for target names

### PR DESCRIPTION
**Pull request by hakonfam**
This is done to allow changing the target name without modifying all
uses of the target.

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -627,35 +630,40 @@ add_custom_command(
...
+add_custom_target(${DRIVER_VALIDATION_H_TARGET} DEPENDS ${DRV_VALIDATION})
 
+set(KOBJ_TYPES_H_TARGET kobj_types_h_target)
 include($ENV{ZEPHYR_BASE}/cmake/kobj.cmake)
 gen_kobj(KOBJ_INCLUDE_PATH)
 
 # Generate offsets.c.obj from offsets.c
 # Generate offsets.h     from offsets.c.obj
 
+set(OFFSETS_LIB offsets)
```
Comment (SebastianBoe) [inline CMakeLists.txt]: Missed a reference:

1202:    add_dependencies(     app_sizing_prebuilt linker_app_sizing_script offsets ${CODE_RELOCATION_DEP} )

**Comment by andyross** [comment]
Curmudgeon review: um.. why?

Build targets are APIs.  They're like function or variable names: standard things with recognizable human language names that developers can read.  You'd never submit a patch turning all function names into macros that can be changed per-build.  (e.g. "Replace all uses of printf() with a PRINTF() macro.  That way the function can be renamed without modifying all callers of the function" -- sounds silly, right?)  What's the value here?  If our target names are bad then we should fix them.

**Comment by SebastianBoe** [comment]
The motivation is to eventually support "variant builds" as a part of this effort: https://github.com/zephyrproject-rtos/zephyr/issues/7868

This patch allows us to change the target name in the future without touching every usage of it.

This is desirable because changes to the target name will be easier to review (O(targets) vs O(target_references) LOC changes).

Changing the target name will be necessary because in a variant build you need to use a different target name for each variant. E.g.

set(ZEPHYR_PREBUILT_EXECUTABLE ${IMAGE_ID}zephyr_prebuilt_executable)

If desirable from a review perspective, we could drop this PR and instead put the commit in the PR that will resolve #7868.

**Review by dbkinder — APPROVED** [review]
the one doc change looks OK, curious that this is the only impact on docs...

**Comment by SebastianBoe** [comment]
recheck

**Review by SebastianBoe — APPROVED** [review]

**Comment by nashif** [comment]
> he motivation is to eventually support "variant builds" as a part of this effort: #7868
> 
> This patch allows us to change the target name in the future without touching every usage of it.
> 
> This is desirable because changes to the target name will be easier to review (O(targets) vs O(target_references) LOC changes).
> 
> Changing the target name will be necessary because in a variant build you need to use a different target name for each variant. E.g.
> 
> set(ZEPHYR_PREBUILT_EXECUTABLE ${IMAGE_ID}zephyr_prebuilt_executable)
> 
> If desirable from a review perspective, we could drop this PR and instead put the commit in the PR that will resolve #7868.

I would like to have this explanation in the commit message to give more context, nobody will know why this change was made once it gets merged, please add more details about WHY we are doing this.

**Review by nashif — CHANGES_REQUESTED** [review]
give more context in commit messages

**Comment by SebastianBoe** [comment]
> I would like to have this explanation in the commit message to give more context, nobody will know why this change was made once it gets merged, please add more details about WHY we are doing this.
-- @nashif 

You are very right about this. This has now been corrected.

**Review by nashif — APPROVED** [review]

**Pull request merged**